src/compat/db_json.cpp

```diff
@@ -979,9 +981,34 @@ db_json_set_func (const JSON_DOC *value, JSON_DOC &doc, const char *raw_path)
       return db_json_er_set_path_does_not_exist (json_pointer_string, &doc);
     }
 
-  // create and insert the value to the specified path
-  p.Create (doc);
-  p.Set (doc, *value, doc.GetAllocator ());
+  const std::string &last_token = json_pointer_string.substr (found + 1);
+
+  if (db_json_path_is_token_valid_array_index (last_token))
```

Comment:
This doesn't look consistent.

We have two cases: parent is non-array or array.

In array case, if 0 index is given, the array is replaced completed. I expect only its first element to be replaced. For non-zero is given, I think we must first check if array index exists. If it does (e.g. index is 1 and array has two or more elements), then element at index should be replaced. If it doesn't exist, only then probably append a new element.

Please reconsider all possible cases and correct the code.